pkg/apis/pipeline/v1beta1/pipeline_types.go

```diff
@@ -296,6 +296,82 @@ func (pt *PipelineTask) validateMatrix(ctx context.Context) (errs *apis.FieldErr
 	return errs
 }
 
+func (pt *PipelineTask) validateExecutionStatusVariablesDisallowed() (errs *apis.FieldError) {
```

Comment:
NIT: will be great to have unit testing for these member functions but we have these covered as a whole in `pipeline_validation_test.go` and hence fine without them